src/components/Web3Wrapper/index.tsx

```diff
@@ -0,0 +1,114 @@
+import { useWeb3React } from '@web3-react/core'
+import { coinbaseWallet, injected, walletConnect } from 'connectors'
+import { Wallet } from 'constants/wallet'
+import usePrevious from 'hooks/usePrevious'
+import { useEffect, useState } from 'react'
+import { useAppDispatch, useAppSelector } from 'state/hooks'
+import { updateWalletOverride } from 'state/user/reducer'
+
+interface ConnectorState {
+  isActive: boolean
+  previousIsActive: boolean | undefined
+  isActivating: boolean
+  isEagerlyConnecting: boolean
+  setIsEagerlyConnecting(connecting: boolean): void
+}
+
+interface Props {
+  children: JSX.Element
+}
+
+const Web3Wrapper = ({ children }: Props) => {
+  const dispatch = useAppDispatch()
+  const { hooks } = useWeb3React()
+  const walletOverrideBackfilled = useAppSelector((state) => state.user.walletOverrideBackfilled)
+
+  const injectedIsActive = hooks.useSelectedIsActive(injected)
+  const coinbaseWalletIsActive = hooks.useSelectedIsActive(coinbaseWallet)
+  const walletConnectIsActive = hooks.useSelectedIsActive(walletConnect)
+
+  const previousInjectedIsActive = usePrevious(injectedIsActive)
+  const previousCoinbaseWalletIsActive = usePrevious(coinbaseWalletIsActive)
+  const previousWalletConnectIsActive = usePrevious(walletConnectIsActive)
+
+  const injectedIsActivating = hooks.useSelectedIsActivating(injected)
+  const coinbaseWalletIsActivating = hooks.useSelectedIsActivating(coinbaseWallet)
+  const walletConnectIsActivating = hooks.useSelectedIsActivating(walletConnect)
+
+  const [isInjectedEagerlyConnecting, setIsInjectedEagerlyConnecting] = useState(false)
+  const [isCoinbaseWalletEagerlyConnecting, setIsCoinbaseWalletEagerlyConnecting] = useState(false)
+  const [isWalletConnectEagerlyConnecting, setIsWalletConnectEagerlyConnecting] = useState(false)
+
+  useEffect(() => {
```

Comment:
Could you define these objects directly? Because you are always defining these objects, I think it is ok to run a forEach and ignore the rule of hooks and omit a lot of the boilerplate - I think that may make it more readable / easier to understand.